doc/vlan_stacking/vlan_stacking_HLD.md

```diff
@@ -0,0 +1,862 @@
+                        }
+                        type leafref {
+                            path /lag:sonic-portchannel/lag:PORTCHANNEL/lag:PORTCHANNEL_LIST/lag:name;
+                        }
+                    }
+                    description "Interface name";
+                }
+
+                leaf s_vlanid {
+                    type leafref {
+                        path /vlan:sonic-vlan/vlan:VLAN/vlan:VLAN_LIST/vlan:vlanid;
+                    }
+                    description "Service VLAN ID";
+                }
+
+                leaf-list c_vlanids {
+                    type union {
+                        type leafref {
+                            path /vlan:sonic-vlan/vlan:VLAN/vlan:VLAN_LIST/vlan:vlanid;
+                        }
+                        type string {
+                            length 1..10;
+                            pattern '[0-9]{1,4}..[0-9]{1,4}';
+                        }
+                    }
+                    description "Customer VLAN ID";
+                }
+
+                leaf s_vlan_priority {
+                    type uint8 {
+                         range 0..7;
+                    }
+                    description "VLAN priority of Service VLAN";
+                }
+             }
+        }
+    }
+}
+```
+
+New yang model `sonic-vlan-translation.yang` is defined to describe VLAN translation configuration.
+
+```
+sonic-vlan-translation {
+
+    yang-version 1.1;
+
+    namespace "http://github.com/Azure/sonic-vlan-translation";
+    prefix vlan-translation;
+
+    import sonic-extension {
+        prefix ext;
+    }
+
+    import sonic-port {
+        prefix port;
+    }
+
+    import sonic-portchannel {
+        prefix lag;
+    }
+
+    import sonic-vlan {
+        prefix vlan;
+    }
+
+    description "VLAN Translation YANG Module for SONiC OS";
+
+    revision 2021-12-15 {
+        description "First Revision";
+    }
+
+    container sonic-vlan-translation {
+        container VLAN_TRANSLATION {
+            description "VLAN_TRANSLATION part of config_db.json";
+
+            list VLAN_TRANSLATION_LIST {
+                key "interface_name s_vlanid";
+
+                leaf interface_name {
+                    type union {
+                        type leafref {
+                            path /port:sonic-port/port:PORT/port:PORT_LIST/port:name;
+                        }
+                        type leafref {
+                            path /lag:sonic-portchannel/lag:PORTCHANNEL/lag:PORTCHANNEL_LIST/lag:name;
+                        }
+                    }
+                    description "Interface name";
+                }
+
+                leaf s_vlanid {
+                    type leafref {
+                        path /vlan:sonic-vlan/vlan:VLAN/vlan:VLAN_LIST/vlan:vlanid;
+                    }
+                    description "Service VLAN ID";
+                }
+
+                leaf c_vlanid {
+                    type leafref {
```

Comment:
What is expected out of leafref here? Are you referring to the vlan id syntax or the existence of c_vlanid at /vlan:sonic-vlan/vlan:VLAN/vlan:VLAN_LIST/vlan:vlanid; ?
c_vlanid will not be known to the system.. and leafref should point to a key.. im not sure, if vlanid is a key here.